ui/core/src/model/secrets.ts

```diff
@@ -25,6 +25,16 @@ export interface Authorization {
   credentialsFile?: string;
 }
 
+export interface OAuth {
+  clientID: string;
+  clientSecret?: string;
+  clientSecretFile?: string;
+  tokenURL: string;
+  scopes?: string[];
+  endpointParams?: Map<string, string[]>;
+  authStyle?: number;
```

Comment:
This range number is not checked in the backend. Could add a check @dougkirkley ?